### ISSUE TITLE
codex_cli: bridged tools silently missing from the first turn on released 0.2.70 (fixed on main, unreleased)

### ISSUE THREAD

**Issue by jfilter** [meridianlabs-ai/inspect_swe#135]
> **Edited after filing:** `main` already fixes this — `codex_mcp_server_config` marks bridged servers `required = true` (#88), plus the pre-launch readiness gate (#106) and configurable startup timeouts (#123). None of that is in a release: the newest PyPI version is 0.2.70 (tagged 9 Aug), and those commits landed after it. So this report stands only for the released version, and the ask is a release rather than a fix. The measurement below is left as confirmation of the mechanism.

### Summary

On `inspect_swe` 0.2.70 (current PyPI release) with `codex_cli(bridged_tools=...)`, the bridged tools intermittently never reach the model: Codex opens its first turn without them, no error is raised, and the sample silently runs as if no tools had been granted. I measured this in 2 of 5 identical runs.

The generated `config.toml` registers the bridge's MCP server as an *optional* server. Codex starts MCP servers concurrently with the first turn and, for an optional server whose startup has not completed, waits only `mcp_optional_startup_grace` — **1 second by default** — before building the turn's tool list without it:

- `codex-rs/codex-mcp/src/connection_manager/tool_catalog.rs`, `capture_binding_with_metadata`: if `startup_complete` is false and `must_wait_for_startup` is false, it waits until the grace deadline and otherwise logs `"omitting pending optional MCP server"` and continues.
- `codex-rs/codex-mcp/src/mcp/mod.rs`: `DEFAULT_OPTIONAL_MCP_STARTUP_GRACE = Duration::from_secs(1)`.
- `codex-rs/codex-mcp/src/connection_manager.rs`: the per-server startup is `join_set.spawn(startup)`; the summary is awaited in a detached task, so session init returns before the server is up.

Whether a run gets its bridged tools therefore depends on whether the MCP startup beats a one-second timer.

### Measurement

Five identical runs, `inspect_swe` 0.2.70, Codex 0.151.0 (`version="auto"`), docker sandbox on macOS/arm64, `mockllm/model`, one trivial host tool. Time from `codex_exec: Sent prompt with event ID` to the turn being dispatched, against whether the bridged tool was in the model request:

| run | sent → turn | bridged tool offered |
|-----|-------------|----------------------|
| 1 | 1.049 s | no |
| 2 | 1.056 s | no |
| 3 | 0.935 s | yes |
| 4 | 0.646 s | yes |
| 5 | 0.733 s | yes |

The split lies exactly on the one-second grace. The MCP handshake itself succeeds in **all five** runs (`rmcp::service: Service initialized as client`), including the two that lose the tools — this is not a transport failure.

Setting `required` from the outside, via `config_overrides={"mcp_servers.<name>.required": "true"}`, gives **5 of 5** runs with the bridged tool. That is the same lever `main` now applies by default, and it works as a stopgap for anyone pinned to the release.

### Ask

Is a release containing #88/#106/#123 planned? Until then every install from PyPI silently loses bridged tools whenever MCP startup crosses one second, and the failure mode is invisible: the run completes, it just measures an agent that had no tools.

### Side note

`config_overrides={"features.experimental_use_rmcp_client": "true"}` — which several examples still recommend for HTTP MCP — is rejected by Codex 0.151.0 with `WARN codex_features: unknown feature key in config: experimental_use_rmcp_client` (visible only with `RUST_LOG=debug`). It is a no-op on current Codex and does not affect the behaviour above.

### Repro

```python
from inspect_ai import Task, task
from inspect_ai.agent import as_solver
from inspect_ai.dataset import Sample
from inspect_ai.tool import tool
from inspect_ai.tool._mcp._tools_bridge import BridgedToolsSpec
from inspect_swe import codex_cli


@tool
def ping():
    async def execute(query: str) -> str:
        """Ask the host something.

        Args:
            query: what to ask
        """
        return f"host answered: {query}"
    return execute


@task
def codex_bridge_race():
    return Task(
        dataset=[Sample(input="Call ping with query=hello", target="OK")],
        solver=as_solver(codex_cli(
            system_prompt="You are a test harness.",
            bridged_tools=[BridgedToolsSpec(name="tools", tools=[ping()])],
            web_search="disabled",
            debug=True,
        )),
        sandbox="docker",
    )
```

Run it a handful of times (`inspect eval codex_bridge_race.py --model mockllm/model`) and inspect the first model event's tool list; a couple of runs will be missing `ping`.
